test/tests.js

```diff
@@ -2042,52 +2223,76 @@ describe( 'passport-saml /', function() {
+            done();
+          } catch (err2) {
+            done(err2);
+          }
         });
       });
       it('errors if idpIssuer is set and wrong issuer', function(done) {
         samlObj.options.idpIssuer = 'foo';
         samlObj.validateRedirect(this.request, function(err) {
-          should.exist(err);
-          err.should.eql(
-            'Unknown SAML issuer. Expected: foo Received: http://localhost:20000/saml2/idp/metadata.php'
-          );
-          done();
+          try {
+            should.exist(err);
+            err.should.eql(
+              'Unknown SAML issuer. Expected: foo Received: http://localhost:20000/saml2/idp/metadata.php'
+            );
+            done();
+          } catch (err2) {
+            done(err2);
+          }
         });
       });
       it('errors if unsuccessful', function(done) {
         this.request = require('./static/sp_slo_redirect_failure');
         samlObj.validateRedirect(this.request, function(err) {
-          should.exist(err);
-          err.should.eql(
-            'Bad status code: urn:oasis:names:tc:SAML:2.0:status:Requester'
-          );
-          done();
+          try {
+            should.exist(err);
+            err.should.eql(
+              'Bad status code: urn:oasis:names:tc:SAML:2.0:status:Requester'
+            );
+            done();
+          } catch(err2) {
+            done(err2);
+          }
         });
       });
       it('errors if InResponseTo is not found', function(done) {
         samlObj.validateRedirect(this.request, function(err) {
-          should.exist(err);
-          err.message.should.eql('InResponseTo is not valid');
-          done();
+          try {
+            should.exist(err);
+            err.message.should.eql('InResponseTo is not valid');
+            done();
+          } catch (err2) {
+            done(err2);
+          }
         });
       });
       it('errors if bad signature', function(done) {
         samlObj.cacheProvider.save('_79db1e7ad12ca1d63e5b', new Date().toISOString(), function(){});
         this.request.Signature = 'foo';
         samlObj.validateRedirect(this.request, function(err) {
-          should.exist(err);
-          err.should.eql('Invalid signature');
-          done();
+          try {
+            should.exist(err);
+            err.should.eql('Invalid signature');
+            done();
+          } catch (err2) {
+            done(err2);
+          }
         });
       });
       it('returns true for valid signature', function(done) {
         samlObj.cacheProvider.save('_79db1e7ad12ca1d63e5b', new Date().toISOString(), function(){});
         samlObj.validateRedirect(this.request, function(err, _data, success) {
-          should.not.exist(err);
-          success.should.eql(true);
-          done();
+          try {
+            should.not.exist(err);
+            success.should.eql(true);
+            done();
+          } catch (err2) {
+            done(err2);
+          }
```

Comment:
This test does not appear to actually be async, so could be converted to a sync test without try/catch.